### PR TITLE
Unify `configure_optimizers` docs

### LINKED ISSUE #7369: Unify return format of `configure_optimizers`
## 🚀 Feature

Unify how optimizers and schedulers are configured in `configure_optimizers`, making the syntax more consistent.

### Motivation

Currently,, in `configure_optimizers` you can do this (code from documentation)

```python
def configure_optimizers(self):
   optimizers = [Adam(...), SGD(...)]
   schedulers = [
      {
         'scheduler': ReduceLROnPlateau(optimizers[0], ...),
         'monitor': 'metric_to_track',
         'interval': 'step',
         'frequency': 10,
         'strict': True,
      },
      LambdaLR(optimizers[1], ...)
   ]
   return optimizers, schedulers
```

or this

```python
def configure_optimizers(self):
   optimizer1 = Adam(...)
   optimizer2 = SGD(...)
   scheduler1 = ReduceLROnPlateau(optimizer1, ...)
   scheduler2 = LambdaLR(optimizer2, ...)
   return (
       {'optimizer': optimizer1, 'lr_scheduler': scheduler1, 'monitor': 'metric_to_track'},
       {'optimizer': optimizer2, 'lr_scheduler': scheduler2},
   )
```

The first returns two lists of the same length, while the second returns a list of dicts, each dict representing an optimizer-scheduler pair.

This seems already like a duplicate, let alone in the first you use `scheduler` as key but in the second you should use `lr_scheduler`.

Besides, scheduler properties such as `interval` can only be specified in the first way, i.e.

```python
   return (
       {'optimizer': optimizer1, 'lr_scheduler': scheduler1, 'monitor': 'metric_to_track', 'interval': 'step'},
       {'optimizer': optimizer2, 'lr_scheduler': scheduler2},
   )
```

won't work (#6948 mentioned this as well), which is very confusing and inconsistent 😕.

### Pitch

As a user, I simply stick to the first way and always returns two lists in `configure_optimizers`. But for the sake of new users, would it be feasible to **deprecate the second way and leave only the first configuring format**?

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Point to `configure_optimizers` in optimizers.rst
Expand and improve docstring

Fixes #7369

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [n/a] Did you write any new necessary tests? (not for typos and docs)
- [n/a] Did you verify new and existing tests pass locally with your changes?
- [n/a] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: pytorch_lightning/core/lightning.py [hunk 1]
```diff
@@ -1116,40 +1116,89 @@ def configure_optimizers(self):
...
             to the number of sequential batches optimized with the specific optimizer.
             It should be given to none or to all of the optimizers.
             There is a difference between passing multiple optimizers in a list,
             and passing multiple optimizers in dictionaries with a frequency of 1:
-            In the former case, all optimizers will operate on the given batch in each optimization step.
-            In the latter, only one optimizer will operate on the given batch at every step.
-            This is different from the ``frequency`` value specified in the lr_dict mentioned below.
+
+            - In the former case, all optimizers will operate on the given batch in each optimization step.
+            - In the latter, only one optimizer will operate on the given batch at every step.
```
Comment (Borda) [inline pytorch_lightning/core/lightning.py:1199]: extra 4 spaces here?
```suggestion
                - In the former case, all optimizers will operate on the given batch in each optimization step.
                - In the latter, only one optimizer will operate on the given batch at every step.
```

**Review by Borda — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
I like it. Some duplicated info gone :)
